tests/lib/schema_validation_test.py

```diff
@@ -170,6 +181,30 @@ def test_valid_without_auto_neg_and_missing_speed(self, default_data):
 
         libnmstate.validator.validate(default_data)
 
+    @parametrize_sriov_valid_total_vfs
+    def test_valid_with_sriov_total_vfs(self, default_data, valid_values):
+        default_data[Interface.KEY][0].update(
+            {
+                Interface.TYPE: InterfaceType.ETHERNET,
+                SRIOV.CONFIG_SUBTREE: {SRIOV.TOTAL_VFS: valid_values},
+            }
+        )
+        libnmstate.validator.validate(default_data)
+
+    @parametrize_sriov_invalid_total_vfs
+    def test_over_maximum_total_vfs_is_invalid(
+        self, default_data, invalid_values
+    ):
+        default_data[Interface.KEY][0].update(
+            {
+                Interface.TYPE: InterfaceType.ETHERNET,
+                SRIOV.CONFIG_SUBTREE: {SRIOV.TOTAL_VFS: invalid_values},
+            }
+        )
+
+        with pytest.raises(js.ValidationError, match=str(invalid_values)):
```

Comment:
not sure it warrants the trouble, but would be better to have it w/ 

```python
invalid_value = 300
max_or_min = 'maximum' if invalid_value > 255 else 'minimum'
comparator = 'greater' if invalid_value > 255 else 'smaller'
with pytest.raises(js.ValidationError, match=f'{invalid_value} is {comparator} than the allowed {max_or_min} of 256'):
    ...
```
Feel free to disregard.

Reply:
This seems to be testing too many internals of jsonschema that Nmstate does not care about (does it @EdDev ?)

Reply:
If we are removing the `maximum` limit then we can simplify this a lot. Thanks!